frontend/src/components/NavigationButtons.tsx

```
@@ -6,33 +7,49 @@ interface NavigationButtonsProps {
+  const showBackButton = useMemo(() => step > 1 && step < 3, [step]);
+  const showNextButton = useMemo(() => step < 3, [step]);
+
   return (
     <div className="flex gap-3 mt-4">
-      {step > 1 && step < 3 && (
+      {showBackButton && (
         <button
-          onClick={onBack}
+          onClick={handleBack}
           className="px-4 py-2.5 h-10 bg-gray-200 text-black rounded-full text-sm font-medium hover:bg-gray-300 transition-colors min-w-[92px]"
         >
```

Biome (2.1.2)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)